src/composer.rs

```diff
@@ -632,7 +632,7 @@ pub trait Composer: Sized + Index<Witness, Output = BlsScalar> {
     /// Asserts `a == b` by appending two gates
     fn assert_equal_point(&mut self, a: WitnessPoint, b: WitnessPoint) {
         self.assert_equal(*a.x(), *b.x());
-        self.assert_equal(*b.y(), *b.y());
```

Comment:
you can avoid these kind of mistakes by deconstructing btw, so that the compiler yells at you if you forget a variable. For example:

```rust
fn assert_equal_point(&mut self, a: WitnessPoint, b: WitnessPoint) {
    let (x1, y1) = a.coords();
    let (x2, y2) = b.coords();
    self.assert_equal(x1, x2);
    self.assert_equal(y1, y2);
}
```

Reply:
unfortunately `a` and `b` are `WitnessPoint`s and don't have a `coord` method.

Reply:
you can implement one, or make x, y public :o